### PR TITLE
Add a script to generate Table of Content (TOC) of MD files

### PR DESCRIPTION
**Pull request by brionmario**
### Purpose

Add script that reads the markdown file, generates the TOC, and updates the file with the TOC.

- Arguments:
    - The file name for which the TOC should be generated.

example 01: node scripts/generate-markdown-toc.js README.md
example 01: node scripts/generate-markdown-toc.js docs/README.md

- Usage:

Add the following placeholders in the markdown file where the TOC should be placed.
```md
<!-- TOS:START - Do not remove or modify this section -->

<!-- TOS:END -->
```
### Related Issues
- N/A

### Related PRs
- N/A

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: scripts/generate-markdown-toc.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ * Add the following placeholders in the markdown file where the TOC should be placed.
+ * <!-- TOS:START - Do not remove or modify this section -->
+ *
+ * <!-- TOS:END -->
+ */
+
+const fs = require("fs");
+const toc = require("markdown-toc");
+
+// eslint-disable-next-line no-console
```
Comment (pavinduLakshan) [inline scripts/generate-markdown-toc.js:39]: We don't need this comment as eslint only catch occurrences of `console.*`.

```suggestion
```

File: scripts/generate-markdown-toc.js
Comment (pavinduLakshan) [inline scripts/generate-markdown-toc.js:1]: Maybe we can integrate this with a github workflow to automatically update TOC in changed markdown files. just a suggestion :)

**Review by pavinduLakshan — DISMISSED** [review]

File: scripts/generate-markdown-toc.js (hunk 1 above)
Comment (brionmario) [reply scripts/generate-markdown-toc.js]: Fixed in https://github.com/wso2/identity-apps/pull/6560/commits/16a9243b7b5014bf57de86cf4e76f53f0be01cd1

File: scripts/generate-markdown-toc.js
Comment (brionmario) [reply scripts/generate-markdown-toc.js:1]: +1

**Review by JayaShakthi97 — APPROVED** [review]

**Pull request merged**